ChainRouter.php

```diff
@@ -306,4 +306,25 @@ public function getRouteCollection()
 
         return $this->routeCollection;
     }
+    
+    /**
+     * Identify if any routers have been added
+     * 
+     * @return bool
+     */
+    public function hasRouters()
+    {
+        return !empty($this->routers);
+    }
+    
+    /**
+     * Get the number of added routers
+     * 
+     * @return integer
+     */
+    public function getNumRouters()
+    {
+        return count($this->routers);
```

Comment:
if you have no routers and call ```all()``` then it will run sortRouters(), call ```krsort()``` and ```array_merge()``` unncessarily. 

We don't want to run the processing for ```all()``` we just need to know ```"are there routers?``` that is all.